### ISSUE TITLE
Local control over relay with physical button

### ISSUE THREAD

**Issue by patrikulus** [Blackymas/NSPanel_HA_Blueprint#215]
TL; DR; additional switch in blueprint config for pairing the physical button with the corresponding relay through ESPHome itself.

Hi,
First, thank you very much for this addon, it's awesome :)

When I create an automation I have "always think about failure" principle in mind. In my home setup, in each room I have primary and secondary light. Primary (left switch on NSPanel) is being connected physically to the relay and those are "dump" bulbs. The secondary (right button) is disconnected from the relay and works as software on/off button for my smart lights. The general idea is that I want to be able to control primary light even when Home Assistant is offline. 

I have installed this blueprint on one of my NSPanels and the problem is that when HA is unavailable (updating, restarting etc) I can't control my primary lights :( Buttons are disconnected from relays by default, which is ok, but I'm missing the option to connect them in the device to be able to operate even without HA.

**Comment by danwright87** [comment]
Hi,
I have all my devices setup in a similar way so if wifi / HA fails lights still work. 

This is something that is already in the marcfager version - https://github.com/marcfager/nspanel-mf, you can handle this on a if not connected, toggle so you can still have automations instead or just always trigger it. 

If you take a copy of the yaml you can edit it and include this, may be possible to add it as a substitution so you have the choice as esphome install time.

**Comment by Blackymas** [comment]
We'll see if we can find a solution for this - if you need it immediately you currently have to do it manually in ESPhome.

**Comment by danwright87** [comment]
i think to be 100% reliable it would have to be done in esphome, as you could set a a sensor and have it store locally but in the case of  after a power cut etc i don't know how reliable that would be.

I've tested the following and it works as expected, maybe as an advanced mode option is best? If you remove the single depth # then it toggles always, if you also remove the double depth #s it only toggles when the nspanel is disconnected.  
Could add it as a local toggle mode setting in substitutions but would need to be included in the base config I believe else would error

`
  ###### LEFT BUTTON BELOW DISPLAY TO TOGGLE RELAY#####
  - platform: gpio
    name: ${device_name} Left Button
    pin:
      number: 14
      inverted: true
    ##### advanced config - uncomment to toggle relay 2 localy. Remove b2 layers of comments to only toggle when HA disconnected ##### 
    #on_click:
    ##   if:
    ##     condition:
    ##       not:
    ##         api.connected:
    ##     then:
    #       - switch.toggle: relay_1


  ##### RIGHT BUTTON BELOW DISPLAY TO TOGGLE RELAY #####
  - platform: gpio
    name: ${device_name} Right Button
    pin:
      number: 27
      inverted: true
    ##### advanced config - uncomment to toggle relay 2 localy. Remove b2 layers of comments to only toggle when HA disconnected ##### 
    #on_click:
    ##   if:
    ##     condition:
    ##       not:
    ##         api.connected:
    ##     then:
    #       - switch.toggle: relay_2

`

let me know if i can do anything to help

**Comment by Blackymas** [comment]
looks good - so in the kind I had also thought or about an additional switch in esphome.

I do not know yet what is the right way.

The logic with the "not api.connected" is a good idea.

Of course you are welcome to help. We are still looking for people who would like to program.

If you are interested we can connect on discord.

**Comment by Blackymas** [comment]
@bkbartk 

We will not bind any HW keys tightly to the relay. 60% of the users use the HW button without relay.

You can put the relay on the HW field in the Blueprint.

in the next update 3.1.0 is an additional fallback in it.

If WiFi or HA would fail, the fallback is activated and you can then directly control the relay via HW buttons and everything works WITHOUT HA and WiFi. thanks @danwright87 

With the update there is no reason to bind the HW buttons to the relay. I use the same with some NSpanel also only a relay and the other HW button I use for something else ;)

**Comment by marcosscriven** [comment]
Can I ask where you get the 60% figure from?

**Comment by patrikulus** [comment]
The fallback option sounds perfect! Thank you all!

**Comment by marcosscriven** [comment]
I do think it's the one thing that I don't think is the correct decision here, and so that's why I'm questioning your assumptions. I don't think an _absence_ of a request for direct relay control means it's not needed. 

In various fora and YouTube comments I've seen multiple people mention this issue. I guess we'll see how well this dynamic switching according to HA availability works.

Certainly I think for some (and I can't claim how many) the relay switches are a family-proof was to turn the lights on and off, no matter what bugs or quirks happen, or if I'm not there.

@OttoWinter of ESPHome wrote something similar on this principle: https://github.com/esphome/esphome-docs/blob/current/guides/automations.rst

> You could write an automation to do this task in Home Assistant's automation engine, but ideally the IoT should work without an internet connection and should not break with the MQTT server being offline.
